### DISCUSSION TITLE
My students don't have permission to run "cmd" thus can't compile their code in Arduino IDE 1.8.x on school PCs

### DISCUSSION THREAD

**Discussion by hogfanboy** [espressif/arduino-esp32#8649] [Ideas]
My school locks the students out of a lot of permissions on the school's PCs. That includes running "cmd"

The students **"can"** compile programs for the ESP8266 using the compiler from ESP8266 community.  
But fails  when compiling with the ESP32, they get an error that there permission group doesn't allow them to run cmd.

So I was wondering if you could take a look on how it works with the ESP8266 boards and apply that to the ESP32 boards

Thanks

**Comment by me-no-dev** [comment]
The requirements for ESP32 and ESP8266 are very different. We use CMD to support different partitions, custom bootloaders and many more things that are not supported on ESP8266. We can not get away from CMD at this point.

**Reply by hogfanboy** [reply]
Thanks for the quick reply.  I'm a high school teacher, and don't have a grasp on writing cores.  But  I hope you can appreciate  there  is a market for using  the ESP32 in educational settings, and schools  often limit permissions of students on schools PCs

**Reply by Xylopyrographer** [reply]
Have you considered [CircuitPython](https://circuitpython.org)? Runs on a number of ESP32 boards.

**Comment by Jason2866** [comment]
@hogfanboy In schools where access is needed to everything it is usual to setup virtual machines on the students machines. The get deleted when PC is powered down and a fresh new Virtual Machine is deployed at start.

**Comment by lisa0987** [comment]
Hi hogfanboy,

It sounds like a challenging situation. If the ESP8266 compiler works without needing "cmd" access, it might be possible to adjust the ESP32 setup similarly. Hopefully, the team can explore this and find a solution. In the meantime, if you're looking for resources, you might want to check out the [Vedu mod APK download ](https://vduapk.com/)for some additional tools that could be useful.

Thanks for bringing this up!
